### ISSUE TITLE
Theorethical Question

### ISSUE THREAD

**Issue by danielstankw** [vincrichard/UnsupervisedAnomalyDetection_Timeseries#3]
Hi,
I wanted to ask you about the concept of LSTM Autoencoder.
If there is only one LSTM in encoder and only one in decoder, then I do not see how a bottleneck is created.
If there is no bottleneck then I do not understand how can a compressed representation of the features can be created.

Would you be so kind to explain this please?

**Comment by vincrichard** [comment]
Hello, it's been a while but I will try my best to answer.
Basically, your bottleneck is the hidden state of your LSTM. During the first step it is quite easy to retrieve all the info and so there is as you said not much of a bottleneck. However, once you have multiple time steps, you are keeping the same hidden state size but you are trying to keep the information about all the timesteps.
Once you arrive at the last timestep N, you encoded information about all previous time steps in the hidden state. At least this is what is enforced during training since we are trying to generate the input timeserie from the last hidden state of the encoder.

Basically we are not reducing the number of features to create a bottleneck, but we are adding information which needs to be stored in the hidden state. In that sense we are reducing the total number of features of the input timeserie to the size of the hidden state.
Hope this makes sense.

**Comment by danielstankw** [comment]
Thank you very much for reply, you helped me a lot :)
